lib/tapioca/dsl/compilers/active_record_typed_store.rb

```diff
@@ -136,9 +135,21 @@ def gather_constants
           T::Hash[Symbol, String],
         )
 
-        sig { params(attr_type: Symbol).returns(String) }
-        def type_for(attr_type)
-          TYPES.fetch(attr_type, "T.untyped")
+        sig { params(field: ActiveRecord::TypedStore::Field).returns(String) }
+        def type_for(field)
+          type = TYPES.fetch(field.type_sym, "T.untyped")
+
+          type = if field.array
+            # `null: false` applies to the array itself, not the elements, which always nilable.
+            # https://github.com/byroot/activerecord-typedstore/blob/2f3fb98/spec/support/models.rb#L46C34-L46C45
+            # https://github.com/byroot/activerecord-typedstore/blob/2f3fb98/spec/active_record/typed_store_spec.rb#L854-L857
+            nilable_element_type = as_nilable_type(type)
+            "T::Array[#{nilable_element_type}]"
+          else
+            type
+          end
+
+          field.null ? as_nilable_type(type) : type
```

Comment:
Small nitpick, do you mind changing this to:
```suggestion
          type = as_nilable_type(type) if field.null
          
          type
```
which IMO is more readable and more idiomatic Ruby than the ternary.